### ISSUE TITLE
Unit tests fail in software renderer only

### ISSUE THREAD

**Issue by chrysn** [openscad/openscad#1413]
There are some tests that are failing for me using the software renderer (libgl1-mesa-swx11 10.6.3-1), but not when using a hardware renderer.

Affected tests are throwntogethertest_issue1089, openscad-cameyeortho_camera-tests and openscad-cameyeortho-viewall_camera-tests.

issue1089 already looks like Z fighting at the expected image. For the camera tests, my best guess is that a frustum is so large that the the projected solid is so slim (low Z range) that even there, Z fighting occurs.

How would this best be resolved?

For the time being, I've blacklisted those tests in the Debian builds.

**Comment by chrysn** [comment]
As github does not allow HTML uploads, attached are screenshots of the test report.

![issue1089](https://cloud.githubusercontent.com/assets/442326/9196940/809b758a-4030-11e5-8c60-00bf4f5e172f.png)
![ortho](https://cloud.githubusercontent.com/assets/442326/9196951/98ba5a8c-4030-11e5-906b-8e404da629e2.png)

**Comment by donbright** [comment]
Your post of the cameye ortho test (the ball with three subtracted cylinders) reminds me a lot of the failures I used to get on a machine with a PPC64 central processing unit running a newer version of Fedora which was using a different software rendering backend based on Gallium. This was a few years ago.

I used to just ignore that failure on that machine, I never did figure out the problem.

**Comment by kintel** [comment]
I agree that this looks like a z buffer resolution issue, probably caused by a combination of a shallow z buffer and bad clipping planes on our side.

**Comment by energypantry** [comment]
Hi! I can take this on with a fast turnaround. Do you accept USDC/USDT (any EVM chain) for payment? If yes, I can start immediately.

**Comment by jordanbrown0** [comment]
Nobody gets paid for this work.
